### DISCUSSION TITLE
Adding complementary documentation to an API type with a custom markdown file

### DISCUSSION THREAD

**Discussion by nockawa** [dotnet/docfx#9456] [Q&A]
I generate documentation for a library, this library implements several custom types which are documented using DocXML in the source code.

But I'd like to add custom documentation for a given type through a dedicated markdown file, so that the generated HTML documentation page for this type contains the DocXML's content, followed by the markdown's one.

My understanding is it has something to do with overwrites, I've tried many combinations but couldn't succeed in having the content of a markdown file appearing in the type's documentation page.

Could someone share some light about how to do that?
Many thanks.

**Comment by paulushub** [comment]
@nockawa Can you provide an illustration or a sample GitHub project to demonstrate your requirements?

**Comment by nockawa** [comment]
Hey @paulushub thanks for reaching, I can do that and even more, giving the answer !
I wanted the documentation generated through DocXML for a given type to also contain a text written separately in markdown.

To achieve something like this:
![image](https://github.com/dotnet/docfx/assets/48723305/a45ef423-4b72-4c89-807c-7191ae6e6c03)

In red square there is the content generated from the .cs file through DocXML
In green square it's the content I've added through an overwrite file, written in markdown.

Things are always simple once you know them...in this case I had to:

Edit the `docfx.json` file like this:
![image](https://github.com/dotnet/docfx/assets/48723305/c31290ca-b892-47b6-aaa5-2a4ae02a158f)

You must exclude the directory you use to store overwrite files from the content and add an overwrite section to define it.

Then the trick is simply putting a markdown file in the overwrite directory and use the right front matter fragment, in my case I've tried all the properties override I could find, using `[*content]` as value, but it never worked as expected.
The only thing you need to do is defining the uid and no property, like this

```markdown
---
uid: Tomate.IMemoryManager
---

## About

`IMemoryManager` is the interface used to implement a custom Memory Manager.
The manager duty is to allocated, resize, free block of memory.
🍅 implements many memory manager, you can find more about it [here](<xref:memory-management-overview>).
```

And now I have the result I was looking for.
